controllers/providers/aws/aws.go

```diff
@@ -507,3 +529,210 @@ func IsStackInConditionState(key string, condition string) bool {
 		return false
 	}
 }
+func IsProfileInConditionState(key *string, condition string) bool {
+	const NONE = "NONE"
```

Comment:
wont it be better to simply take `key` as a string (not ptr)? since `ResourceState` is coming from AWS, all getters around such structs should probably give out copies and not a pointer. if we do `return aws.StringValue(x)` from the getter we can get values instead of pointers.

also you should not required a clause for NONE here, because `StateDiscovery` (which runs early), checks `if state.IsProvisioned()` and does the checking of whether deleted if it's false.

We should probably do work across all types to make sure we only get pointers for structs we actively modify (such as the instance group spec).

It should be OK to address this in a future PR

Reply:
This is great.  I'd love to have a general rule for when to use references or values.

Reply:
Changed my mind based upon your comment below.  When I move the state logic to CloudDiscovery, I will do the NIL mapping there.